scripts/cnode-helper-scripts/blockLog.sh

```diff
@@ -0,0 +1,122 @@
+	[[ "$(jq -r .TraceBlockFetchClient "${CONFIG}")" != "true" ]] && echo -e "${RED}Error:${NC} In config file please set ${FG_YELLOW}\"TraceBlockFetchClient\":\"true\"${NC}" && errors=1
+	[[ "$(jq -r .TraceChainSyncHeaderServer "${CONFIG}")" != "true" ]] && echo -e "${RED}Error:${NC} In config file please set ${FG_YELLOW}\"TraceChainSyncHeaderServer\":\"true\"${NC}" && errors=1
+	[[ $errors -eq 1 ]] && exit 1
+else 
+	echo -e "${RED}Error:${NC} Failed to locate json configuration file" && exit 1
+fi
+
+echo "$(date) parsing ${logfile} ..." | tee ${CNODE_HOME}/logs/blockLog_debug.log
+
+blockHeightPrev=$(curl -s -H 'Accept: application/json' http://127.0.0.1:$(cat $CONFIG | jq .hasEKG)/ | jq -r .cardano.node.metrics.blockNum.int.val)
+blockTimeDeltaSlots=0
+
+getDeltaMS() {
+	echo $(echo "$2 $4" | awk -F '[:, ]' '{print ($1*3600000+$2*60000+$3*1000+$4)-($5*3600000+$6*60000+$7*1000+$8) }')
+}
+
+while true
+do
+	blockHeight=$(curl -s -H 'Accept: application/json' http://127.0.0.1:$(cat $CONFIG | jq .hasEKG)/ | jq -r .cardano.node.metrics.blockNum.int.val)
+	blockTimeDeltaSlots=$((blockTimeDeltaSlots+1))
+	#echo "$(date) CP0 deltaSlots:${blockTimeDeltaSlots}  blockheight:${blockHeight} ${blockHeightPrev}"
+	if [ -z $blockHeight ] || [ "$blockHeight" == "null" ] || [ "$blockHeightPrev" == "null" ] ; then
+		echo "WARN: can't query EKG on http://127.0.0.1:$(cat $CONFIG | jq .hasEKG)/ ..."
+	elif  [ "$blockHeight" -gt "$blockHeightPrev" ] ; then # new Block
+	
+		#echo "$(date) CP1 deltaSlots:${blockTimeDeltaSlots}  blockheight:${blockHeight} ${blockHeightPrev}" | tee -a ${CNODE_HOME}/logs/blockLog_debug.log
+		blockHash=$(grep -m 1 "$blockHeight" ${logfile} | jq -r .data.block)  # we parse the second last, now fully logged block
+		blockLog=$(grep $blockHash ${logfile})
+
+		if [[ ! -z "$blockLog" ]]; then
+			blockLogLineCycles=0
+			while IFS= read -r blockLogLine;do
+				# parse block and propagation metrics from different log kinds
+				lineKind=$(jq -r .data.kind <<< $blockLogLine)
+				case $lineKind in
+					ChainSyncClientEvent.TraceDownloadedHeader)
+						[ -z "$blockTimeTbh" ] && blockTimeTbh=$(date -d $(jq -r .at <<< $blockLogLine) +"%F %T,%3N")
+						[ -z "$blockSlot" ] && blockSlot=$(jq -r .data.slot <<< $blockLogLine)
+						[ -z "$blockSlotTime" ] && blockSlotTime=$(getDateFromSlot ${blockSlot} '%(%F %T)T')
+						;;
+					SendFetchRequest)
+						[ -z "$blockTimeSfr" ] && blockTimeSfr=$(date -d $(jq -r .at <<< $blockLogLine) +"%F %T,%3N")
+						;;
+					CompletedBlockFetch)
+						[ -z "$blockTimeCbf" ] && blockTimeCbf=$(date -d $(jq -r .at <<< $blockLogLine) +"%F %T,%3N")
+						[ -z "$blockTimeCbfAddr" ] && blockTimeCbfAddr=$(jq -r .data.peer.remote.addr <<< $blockLogLine)
+						[ -z "$blockTimeCbfPort" ] && blockTimeCbfPort=$(jq -r .data.peer.remote.port <<< $blockLogLine)
+						[ -z "$blockDelay" ] && blockDelay=$(jq -r .data.delay <<< $blockLogLine)
+						[ -z "$blockSize" ] && blockSize=$(jq -r .data.size <<< $blockLogLine)
```

Comment:
Any way to reduce repetitive external jq calls helps to reduce resource imprint the script has. I noticed this with gLiveView, that a single call that grabs all values is better. Then let bash deal with splitting it into different variables if needed. See getNodeMetrics() in env. It grabs all values and pipes it out as tsv, then this is split into an array so each result in array contain selected values. 
https://github.com/cardano-community/guild-operators/blob/master/scripts/cnode-helper-scripts/env#L466

Reply:
I implemented it like this